xcube/genl2c/cli.py

```diff
@@ -107,44 +119,6 @@ def main(args: Optional[List[str]] = None):
 def _handle_error(e, traceback_mode):
     print(f'error: {e}', file=sys.stderr)
     if traceback_mode:
+        import traceback
         traceback.print_exc(file=sys.stderr)
     return 2
-
-
-class GenL2CHelpFormatter(argparse.HelpFormatter):
-
-    def format_help(self):
-        # noinspection PyUnresolvedReferences
-        help_text = super().format_help()
-
-        input_processors = get_obj_registry().get_all(type=InputProcessor)
-        output_writers = query_dataset_io(lambda ds_io: 'w' in ds_io.modes)
-
-        help_text += '\ninput processors to be used with option --proc:\n'
-        help_text += self._format_input_processors(input_processors)
-        help_text += '\noutput formats to be used with option --writer:\n'
-        help_text += self._format_dataset_ios(output_writers)
-        help_text += '\n'
-
-        return help_text
-
-    @classmethod
-    def _format_input_processors(cls, input_processors):
-        help_text = ''
-        for input_processor in input_processors:
-            fill = ' ' * (34 - len(input_processor.name))
-            help_text += f'  {input_processor.name}{fill}{input_processor.description}\n'
-        return help_text
-
-    @classmethod
-    def _format_dataset_ios(cls, dataset_ios):
-        help_text = ''
-        for ds_io in dataset_ios:
-            fill1 = ' ' * (24 - len(ds_io.name))
-            fill2 = ' ' * (10 - len(ds_io.ext))
-            help_text += f'  {ds_io.name}{fill1}(*.{ds_io.ext}){fill2}{ds_io.description}\n'
-        return help_text
-
-
-if __name__ == '__main__':
-    sys.exit(main())
```

Comment:
This code shall not be thrown away, but used differently. Find out how to display this info using click.